event/src/event.rs

```diff
@@ -0,0 +1,100 @@
+use crate::args::UnsignedEvent;
+use crate::DidDocument;
+use anyhow::Result;
+use ceramic_core::{Cid, DagCborEncoded, Jwk, Jws};
+use multihash::{Code, MultihashDigest};
+use serde::Serialize;
+
+// https://github.com/multiformats/multicodec/blob/master/table.csv
+const DAG_CBOR_CODEC: u64 = 0x71;
+
+pub struct Event {
+    pub cid: Cid,
+    pub linked_block: DagCborEncoded,
+    pub jws: Jws,
+}
+
+impl Event {
+    pub async fn new<'a, T: Serialize>(
+        unsigned: &'a UnsignedEvent<'a, T>,
+        signer: &'a DidDocument,
+        jwk: &'a Jwk,
+    ) -> Result<Self> {
+        // encode our commit with dag cbor, hashing that to create cid
+        let linked_block = DagCborEncoded::new(&unsigned)?;
+        let cid = Cid::new_v1(DAG_CBOR_CODEC, Code::Sha2_256.digest(linked_block.as_ref()));
+        let jws = Jws::new(jwk, signer, &cid)?;
+        Ok(Self {
+            cid,
+            linked_block,
+            jws,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{DidDocument, EventArgs, StreamId};
+    use std::str::FromStr;
+
+    #[test]
+    fn should_roundtrip_json_data() {
+        let did_str = "some_did";
+        let data = serde_json::json!({
+            "creator": did_str,
+            "radius": 1,
+            "red": 2,
+            "green": 3,
+            "blue": 4,
+        });
+        let encoded = DagCborEncoded::new(&data).unwrap();
+        let decoded: serde_json::Value = serde_ipld_dagcbor::from_slice(encoded.as_ref()).unwrap();
+        assert_eq!(decoded, data);
+    }
+
+    #[tokio::test]
+    async fn should_roundtrip_init_event() {
+        let did = DidDocument::new("did:key:blah");
+        let model =
+            StreamId::from_str("kjzl6kcym7w8y6of44g27v981fuutovbrnlw2ifbf8n26j2t4g5mmm6zc43nx1u")
+                .unwrap();
+        let args = EventArgs::new_with_parent(&did, &model);
+        let evt = args.init().await.unwrap();
+        let _: serde_json::Value = serde_ipld_dagcbor::from_slice(evt.encoded.as_ref()).unwrap();
+    }
```

Comment:
Should this test have some assertions?

Reply:
We could change the unwrap to assert not throw, but essentially doing assertion by unwrap.

Reply:
Makes sense now